ext/oembed/ang/crmOembedSharing/main.js

```diff
@@ -0,0 +1,41 @@
+(function(angular, $, _) {
+  // Example usage: <crm oembed-sharing page="https://example.com/foo/bar?id=123"></div>
+  angular.module('crmOembedSharing').directive('crmOembedSharing', function() {
+    return {
+      restrict: 'AE',
```

Comment:
Another newbie question: what does AE mean or do in this context?

Reply:
A directive can be activated based on HTML (a)ttributes (`<div crm-oembed-sharing>` or HTML (e)lements (`<crm-oembed-sharing />`) or HTML (c)lasses (`<div class="crm-oembed-sharing">`). 

The `exports` and `restrict` both speak to this. It's the same "A". Slightly different angles:

*  The `exports` is more of a build-instruction/dependency-info. If somebody else wants to use`crm-oembed-sharing`, then we should autoload the module.
* The `restrict` is a runtime thing -- it tells the browser how to target things at run time (eg how to use DOM selectors).
 
This mismatch between them (`exports=A`, `restrict:AE`) is a latent bug.